### ISSUE TITLE
Van-event weapon spawns deleting themselves with DeleteUnwantedWeapon

### ISSUE THREAD

**Issue by Shadowysn** [orinuse/L4D2_default_map_addons#10]
[L113](https://github.com/orinuse/L4D2_default_map_addons/blob/main/coop/MP-DeadAirTerminals/packing/scripts/vscripts/entitygroups/patch_da_weaponsgroup.nut#L113)
[L412](https://github.com/orinuse/L4D2_default_map_addons/blob/main/coop/MP-DeadAirTerminals/packing/scripts/vscripts/entitygroups/patch_da_weaponsgroup.nut#L412)
[L523](https://github.com/orinuse/L4D2_default_map_addons/blob/main/coop/MP-DeadAirTerminals/packing/scripts/vscripts/entitygroups/patch_da_weaponsgroup.nut#L523)
These `patch_da_weaponsgroup.nut` weapon spawns are removing themselves the moment they spawn.
Odd, I could vaguely remember they spawned the first time I tested the mod.

Another unrelated bug found when I was testing this bug was the two gascan-propane-oxygen tank spawns next to the boxes spawn non-physics-affected melees which float in mid-air. Can be fixed by setting **spawnflags** to **1**. The gascan-propane-oxygen tanks are still affected by physics anyway without the spawnflag set.
Luckily, it appears the gascans won't delete themselves.
![meleebug](https://user-images.githubusercontent.com/44493487/212614228-79dac6c4-1f07-47a7-ba7e-bd0829452ed6.png)

**Comment by orinuse** [comment]
> [L113](https://github.com/orinuse/L4D2_default_map_addons/blob/main/coop/MP-DeadAirTerminals/packing/scripts/vscripts/entitygroups/patch_da_weaponsgroup.nut#L113)
> [L412](https://github.com/orinuse/L4D2_default_map_addons/blob/main/coop/MP-DeadAirTerminals/packing/scripts/vscripts/entitygroups/patch_da_weaponsgroup.nut#L412)
> [L523](https://github.com/orinuse/L4D2_default_map_addons/blob/main/coop/MP-DeadAirTerminals/packing/scripts/vscripts/entitygroups/patch_da_weaponsgroup.nut#L523)
> These `patch_da_weaponsgroup.nut` weapon spawns are removing themselves the moment they spawn.

Is that so? Could you provide the whereabouts of these weapon spawners and your method of identifying the issue? I don't think I'm able to really grasp how is it going wrong here.

> Another unrelated bug found when I was testing this bug was the two gascan-propane-oxygen tank spawns next to the boxes spawn non-physics-affected melees which float in mid-air. Can be fixed by setting **spawnflags** to **1**. The gascan-propane-oxygen tanks are still affected by physics anyway without the spawnflag set.

Sounds like you're describing [`orin_weapons_patch14`](https://github.com/orinuse/L4D2_default_map_addons/blob/main/coop/MP-DeadAirTerminals/packing/scripts/vscripts/entitygroups/patch_da_weaponsgroup.nut#L440), though I'd say its better to remove spawning of melees from the entity.

**Comment by Shadowysn** [comment]
> Is that so? Could you provide the whereabouts of these weapon spawners and your method of identifying the issue? I don't think I'm able to really grasp how is it going wrong here.

Played and restarted the level two times and checking the spawns that have **DeleteUnwantedWeapon**, with the **DeleteUnwantedWeapon** function commented out first time and then reactivated the second time.
![delunwep1](https://user-images.githubusercontent.com/44493487/212624900-24c79253-dcff-4993-9a90-5dbb8f1c4885.png)
![delunwep2](https://user-images.githubusercontent.com/44493487/212624913-eec21bcf-510c-4c15-8588-36ae2176ed21.png)

> Sounds like you're describing [`orin_weapons_patch14`](https://github.com/orinuse/L4D2_default_map_addons/blob/main/coop/MP-DeadAirTerminals/packing/scripts/vscripts/entitygroups/patch_da_weaponsgroup.nut#L440), though I'd say its better to remove spawning of melees from the entity.

Oh, so it's an accidental keyvalue?
I'd edit the entity group but I want to consider the vmfs the entitygroups are compiled from.

**Comment by orinuse** [comment]
> Played and restarted the level two times and checking the spawns that have **DeleteUnwantedWeapon**, with the **DeleteUnwantedWeapon** function commented out first time and then reactivated the second time.

If its identification through pure visual, then I fear it may be inaccurate - The director populates select few spawns of the same type using its density settings, also I do not know if the entity being deleted is a different **entity**. Nonetheless, I have no say and will sleep on this for moe.

> Oh, so it's an accidental keyvalue?

It most likely is intentional, however whether it has been thoroughly tested or not is a mystery.

> I'd edit the entity group but I want to consider the vmfs the entitygroups are compiled from.

Unfortnately these entity group VMFs are very outdated and aren't too reliable anymore, but I could merge it all into one VMF in the future.

**Comment by Shadowysn** [comment]
Oops. I just realized I got the original weapon spawns confused for being the proper new spawn items. My bad, this issue is now mostly invalid 😅
Still odd that the new `weapon_spawn`s aren't spawning at all for me, even after I've restarted it many times.
But while in the process of realizing this, I found an oddity with the `weapon_spawn` items that contributed to my confusion and this issue for the wrong thing.
